File: 卡码/代码随想录/Java/src/com/kurumi/codeThoughts/字符串/_02_反转字符串2/LeetCode_0541.java
package com.kurumi.codeThoughts.字符串._02_反转字符串2;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 541. 反转字符串 II
 */
public class LeetCode_0541 {
    static class Solution {
        public String reverseStr(String s, int k) {
            char[] ch = s.toCharArray();
            for (int i = 0; i < ch.length; i += 2 * k) {
                int start = i, end = Math.min(start + k - 1, ch.length - 1);
                while (start < end) {
                    ch[start] ^= ch[end];
                    ch[end] ^= ch[start];
                    ch[start++] ^= ch[end--];
                }
            }
            return new String(ch);
        }
    }
}

File: 卡码/代码随想录/C++/字符串/02-反转字符串2/LeetCode_0541.cpp
//541. ·´×ª×Ö·û´® II
#include <iostream>
#include <string>
#include <algorithm>
using namespace std;

class Solution {
public:
  string reverseStr(string s, int k) {
  	int n=s.length();
		for(int i=0;i<s.length();i+=2*k){
			reverse(s.begin()+i,s.begin()+min(i+k,n));
		}
		return s;
  }
};
